pydropsonde/circles.py

```diff
@@ -16,5 +21,138 @@ class Circle:
     segment_id: str
 
     def dummy_circle_function(self):
-        print(self.flight_id, self.segment_id)
+        print(self.flight_id, self.segment_id, self.platform_id)
+        return self
+
+    def get_xy_coords_for_circles(self):
+        x_coor = (
+            self.circle_ds.lon * 111.320 * np.cos(np.radians(self.circle_ds.lat)) * 1000
+        )
+        y_coor = self.circle_ds.lat * 110.54 * 1000
+
+        # converting from lat, lon to coordinates in metre from (0,0).
+
+        c_xc = np.full(np.size(x_coor, 1), np.nan)
+        c_yc = np.full(np.size(x_coor, 1), np.nan)
+        c_r = np.full(np.size(x_coor, 1), np.nan)
+
+        for j in range(np.size(x_coor, 1)):
+            a = ~np.isnan(x_coor.values[:, j])
+            if a.sum() > 4:
+                c_xc[j], c_yc[j], c_r[j], _ = cf.least_squares_circle(
+                    [
+                        (x, y)
+                        for x, y in zip(x_coor.values[:, j], y_coor.values[:, j])
+                        if ~np.isnan(x)
+                    ]
+                )
+
+        with warnings.catch_warnings():
```

Comment:
any reason why you catch this warning here? I probably just don't know what it's for, but usually warnings are there for a reason and suppressing them should be done carefully (but probably you know, why it occurs).

Reply:
ah... it probably occurs because the example circles are too short in length... but maybe we should then just add more and catch cases where the fit does not work?